### ISSUE TITLE
Decide whether non-location permissions should be optional or required

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-docs#700]
We now require multiple permissions in addition to the location. In the UI, we currently enforce that the user gives all the permissions up front, but they can turn them off at any time.

The location permission is clearly needed, but we can choose to treat the other permissions as optional. If they are not present, our automated algorithms will not work as well, and important issues may be missed. Here's what should happen if each data type is not present:
- motion activity: all trips will have the sensor-based inference = UNKNOWN. But the user labels and label assist will still work.
- notifications: user will miss notifications about issues with the tracking. But if they open the app every day, they will know whether it works on not anyway. user will also miss notifications from program admins, but maybe they don't care about those.
- background restrictions: again, if the user does commits to launching the app every day, then it will never get into the "unused apps" state and they don't need to remove the restriction

So it looks like keeping them all optional would work if the user is willing to do some manual work down the road. 

Pro: It would be more in line with android best practices.
Con: Testing is potentially much more complicated because we need to make sure that all the combinations work properly instead of only testing the one true way.

**Comment by shankari** [comment]
@lgharib @PatGendre @asiripanich any thoughts on whether the non-location permissions should be required or optional?

- they are actually optional - if the user has notifications turned off on iOS, everything will still work, for example
- but they are strongly recommended, and most of our testing has been with them enabled.
   - If the motion activity is disabled, for example, then there is no automated mode detection.
   - But maybe that is OK if people want to label all their trips.
   - But then even the label assist may be worse if we use the sensed mode as an input to the labeling algorithm

In other words, requiring everything results in a simpler system and makes things easier for us.
Making everything other than location optional results in a more complex and configurable system and may be easier for the end users.

Thoughts?

**Comment by PatGendre** [comment]
@shankari I would be in favor of making it required, as anyway location is already required (and background execution, too?), so the user has to explicitly accept permissions at onboarding.

**Comment by shankari** [comment]
@asiripanich would it be bad if the motion activity was required as well? As I outline above, making everything required reduces the complexity of the system and leads to fewer questions along the lines of "why is this not working?"

**Comment by asiripanich** [comment]
> @asiripanich
> 
> wrt mode detection accuracy, it seems to be fairly accurate in France.
> 
> The pipeline has been primarily tuned based on the SF Bay Area since that is where all the ground truth data is from.
> 
> Would you be willing to collect data using a principled approach like MobilityNet so that we can:
> 
>     * evaluate the analysis pipeline against it
> 
>     * potentially experiment with improving it so it handles Australia better
> 
> 
> This shouldn't be very complicated - you should be able to hire an undergrad to do the work.
> 
>     * Specifying the routes for MobilityNet primarily requires local knowledge, and
> 
>     * collecting data using the MobilityNet approach doesn't require any technical expertise, only the ability to follow instructions that show up on the phone

Yes, I do realise that the model was tuned using Bay area data. So, no complaints here. 
MobilityNet sounds interesting. Could you point to the paper? or the complete instructions on MobilityNet.
